src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/EntityFrameworkServices.cs

```diff
@@ -468,23 +390,25 @@ private ModelMetadata GetModelMetadata(Type dbContextType, Type modelType, Model
             return new ModelMetadata(entityType, dbContextType);
         }
 
-        private DbContext TryCreateContextUsingAppCode(Type dbContextType, ModelType startupType)
+        private DbContext TryCreateContextUsingAppCode(Type dbContextType, Type startupType)
         {
-            try {
-                var builder = new WebHostBuilder();
-                builder.UseKestrel()
-                        .UseContentRoot(Directory.GetCurrentDirectory());
+            try
+            {
+                // Use EF design APIs to get the DBContext instance.
+                var operationHandler = new OperationReportHandler();
+                var operationReporter = new OperationReporter(operationHandler);
+                var startupInvoker = new StartupInvoker(
```

Comment:
Hmm, there is a higher-level method that might be better: `DbContextOperations.CreateContext()`. This would account for `DbContext` types who's design-time creation has been overridden using `IDbContextFactory`.